deltacat/compute/compactor_v2/private/compaction_utils.py

```diff
@@ -173,6 +171,33 @@ def _build_uniform_deltas(
     )
 
 
+def _group_uniform_deltas(params: CompactPartitionParams, uniform_deltas):
+    num_deltas = len(uniform_deltas)
+    num_rounds = params.num_rounds
+    if num_rounds == 1:
+        return [uniform_deltas]
+    assert (
+        num_rounds <= num_deltas
+    ), f"{params.num_rounds} rounds should be less than the number of uniform deltas, which is {len(uniform_deltas)}"
+    assert (
+        params.hash_bucket_count != 1
+    ), "Hash bucket count is set to 1, we should not be grouping uniform deltas into batches"
+    if num_rounds >= num_deltas:
+        return [uniform_deltas]
+    sublist_length = num_deltas // num_rounds
+    uniform_deltas_grouped = []
+    start_index = 0
+    for _ in range(num_rounds):
+        end_index = start_index + sublist_length
+        sublist = uniform_deltas[start_index:end_index]
+        uniform_deltas_grouped.append(sublist)
+        start_index = end_index
+    remaining_deltas = uniform_deltas[start_index:]
+    if remaining_deltas:
+        uniform_deltas_grouped[-1].extend(remaining_deltas)
```

Comment:
Looks like this split logic will result in N+1 rounds?

Reply:
Oh I see, you're just extending the last piece of uniform_deltas_group, so should still be N rounds.